### PR TITLE
fix accessibility challenge: tabindex="0" -> tabindex="-1"

### LINKED ISSUE #42311: Bad example in Applied Accessibility/Use tabindex to Add Keyboard Focus to an Element
**Affected page**
https://www.freecodecamp.org/learn/responsive-web-design/applied-accessibility/use-tabindex-to-add-keyboard-focus-to-an-element

**Issue**
The instructions in this lesson are:

"He knows input fields automatically get keyboard focus, but **he wants to make sure his keyboard users pause at the instructions while tabbing through the items**. Add a tabindex attribute to the p tag and set its value to 0."

This is a big accessibility anti-pattern. In general, tabindex="0" should only be set on elements that are interactive. There are other more accessible options for associating the instructions with the form. First, you would want to move the instructions just before the opening form element to ensure they are read before a screen reader goes into forms mode. You might even associate the instructions with the first input field in the form using aria-describedby but I think a better method would be to indicate which inputs are required individually using one of several options I won't go into here.

Regardless of how you ultimately handle this issue, the one way you do not want to do it is to add tabindex="0" to the p tag in order to insert that paragraph into the natural tab order. 

**References**
https://www.w3.org/TR/html53/editing.html#the-tabindex-attribute
https://www.w3.org/WAI/tutorials/forms/instructions/
https://www.a11yproject.com/posts/2021-01-28-how-to-use-the-tabindex-attribute/

**Suggested Alternative Example**
Generally you would use tabindex="0" when creating custom interactive components, which might be too much to explain for this lesson? I suppose you could use a simpler example that turns a div into a link (or something similar) but then not only do you need to add tabindex but you also need to add a role and explain that you have to use javascript to implement all of the functionality that a link would normally have and also explain that most accessibility experts would recommend you always use an actual anchor tag instead of trying to turn a div into something it's not.

One example that does come to mind is setting tabindex="0" on an element that will scroll when it overflows. Some browsers don't automatically add these elements to the tab order and thus keyboard users can't scroll them without the added tabindex. For an example see https://adrianroselli.com/2020/11/under-engineered-responsive-tables.html.

Another option might be to use tabindex="-1". This is often used to send focus to a non-interactive element. For example, we use it for quizzing, when the student submits their answer we remove the submit button and set the focus on the element that contains the feedback (which has tabindex="-1" set on it).

### PR DESCRIPTION
**Pull request by hchiam**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #42311

I framed the rationale/motivation as spam bot detection.

### CODE REVIEW

**Review by hchiam — COMMENTED** [review]
Regular scanning with a screen reader can still get to it. I'll add `aria-hidden="true"`.

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-tabindex-to-add-keyboard-focus-to-an-element.md [hunk 1]
```diff
@@ -107,9 +122,12 @@ assert($('p').attr('tabindex') == '0');
   </header>
   <section>
     <form>
+      <p>Instructions: Fill in ALL your information then click <b>Submit</b></p>
+      <label for="username">Username:</label>
+      <input type="text" id="username" name="username"><br>
 
 
-      <p tabindex="0">Instructions: Fill in ALL your information then click <b>Submit</b></p>
+      <input tabindex="-1" type="text" id="hiddenInput" aria-label="hidden input to catch spam bots" aria-hidden="true" placeholder="Don't fill this in!">
```
Comment (ShaunSHamilton) [inline curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-tabindex-to-add-keyboard-focus-to-an-element.md:130]:
```suggestion
      <input tabindex="-1" type="text" id="hidden-input" aria-label="hidden input to catch spam bots" aria-hidden="true" placeholder="Don't fill this in!">
```

**Review by ShaunSHamilton — COMMENTED** [review]
Just for discussion:

Have we ever decided on a convention for HTML/CSS naming? (e.g. [Google's HTML/CSS Style Guide](https://google.github.io/styleguide/htmlcssguide.html#ID_and_Class_Name_Delimiters))

File: curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-tabindex-to-add-keyboard-focus-to-an-element.md (hunk 1 above)
Comment (hchiam) [reply curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-tabindex-to-add-keyboard-focus-to-an-element.md]: Thanks @ShaunSHamilton, I've updated it to use kebab-case.

Comment (hchiam) [reply curriculum/challenges/english/01-responsive-web-design/applied-accessibility/use-tabindex-to-add-keyboard-focus-to-an-element.md]: When I look at [id's in other markdown files](https://github.com/freeCodeCamp/freeCodeCamp/search?l=Markdown&o=desc&p=2&q=id%5C%3D%5C%22&s=) and use Ctrl+F on the results page for `id="`, I see a mix of camelCase and kebab-case. So far it looks like just `id="xpText"` is using camelCase, and most of the unique multi-word IDs are already using kebab-case (e.g. `id="header-img"` and `id="calorie-form"`).

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Some things about these changes:
- If the title changes, it needs to change in the meta holding the challenge order to match.
- We need to discuss if we should also change the filename - we have done this in the past, so I assume this is the best way to go
- I am not a fan of introducing this as a way to _trick_ "spam bots". I suggest we discuss a bit more on the issue how to implement something more similar to what @bbsmooth suggested.

**Comment by hchiam** [comment]
@ShaunSHamilton feel free to close this PR or take it over. I'm back on the fence about whether an example with `tabindex="-1"` or `tabindex="0"` would be simpler to set up and explain to a beginner.

**Comment by raisedadead** [comment]
Hey @ShaunSHamilton what's our direction with this?

**Comment by ShaunSHamilton** [comment]
@raisedadead It looks like we should close this PR to continue discussion on the issue itself, and come to a conclusion as to how to teach the topic.

---

> @ShaunSHamilton feel free to close this PR or take it over.

@hchiam Thank you, for your time on this. I am going to close this, with the expectation that we will come to a conclusion as how to approach this topic on the issue.

**Pull request closed without merging**